cpp/models/ode_secirvvs/parameters_io.h

```diff
@@ -753,636 +835,82 @@ IOResult<void> set_vaccination_data(std::vector<Model<FP>>& model, const std::st
-                model[county].parameters.template get<TimeInfectedNoSymptoms<double>>()[(AgeGroup)group] * reduc_t)));
-            t_InfectedSymptoms_fv[county].push_back(static_cast<int>(std::round(
-                model[county].parameters.template get<TimeInfectedSymptoms<double>>()[(AgeGroup)group] * reduc_t)));
-            t_InfectedSevere_fv[county].push_back(static_cast<int>(
-                std::round(model[county].parameters.template get<TimeInfectedSevere<double>>()[(AgeGroup)group])));
-            t_InfectedCritical_fv[county].push_back(static_cast<int>(
-                std::round(model[county].parameters.template get<TimeInfectedCritical<double>>()[(AgeGroup)group])));
 
-            double reduc_immune_exp =
-                model[county].parameters.template get<ReducExposedImprovedImmunity<double>>()[(AgeGroup)group];
-            double reduc_immune_inf =
-                model[county].parameters.template get<ReducInfectedSymptomsImprovedImmunity<double>>()[(AgeGroup)group];
-            double reduc_immune_hosp =
-                model[county].parameters.template get<ReducInfectedSevereCriticalDeadImprovedImmunity<double>>()[(
-                    AgeGroup)group];
-            double reduc_immune_icu =
-                model[county].parameters.template get<ReducInfectedSevereCriticalDeadImprovedImmunity<double>>()[(
-                    AgeGroup)group];
-            mu_C_R_fv[county].push_back((
-                1 - reduc_immune_inf / reduc_immune_exp *
-                        (1 - model[county]
-                                 .parameters.template get<RecoveredPerInfectedNoSymptoms<double>>()[(AgeGroup)group])));
-            mu_I_H_fv[county].push_back(
-                reduc_immune_hosp / reduc_immune_inf *
-                model[county].parameters.template get<SeverePerInfectedSymptoms<double>>()[(AgeGroup)group]);
-            // transfer from H to U, D unchanged.
-            mu_H_U_fv[county].push_back(
-                reduc_immune_icu / reduc_immune_hosp *
-                model[county].parameters.template get<CriticalPerSevere<double>>()[(AgeGroup)group]);
-
-            sum_mu_I_U_fv[county] +=
-                reduc_immune_icu / reduc_immune_hosp *
-                model[county].parameters.template get<CriticalPerSevere<double>>()[AgeGroup(group)] *
-                reduc_immune_hosp / reduc_immune_inf *
-                model[county].parameters.template get<SeverePerInfectedSymptoms<double>>()[AgeGroup(group)];
-            mu_I_U_fv[county].push_back(
-                reduc_immune_icu / reduc_immune_hosp *
-                model[county].parameters.template get<CriticalPerSevere<double>>()[AgeGroup(group)] *
-                reduc_immune_hosp / reduc_immune_inf *
-                model[county].parameters.template get<SeverePerInfectedSymptoms<double>>()[AgeGroup(group)]);
+        // TODO: Reuse more code, e.g., set_divi_data (in secir) and a set_divi_data (here) only need a different ModelType.
```

Comment:
Is this already an Issue? That shouldnt be hard to do and saves a lot of duplication

Reply:
Opened an issue #1108